src/middlewared/middlewared/plugins/config.py

```diff
@@ -147,6 +148,43 @@ def __upload(self, config_file_name):
         # Now we must run the migrate operation in the case the db is older
         open(NEED_UPDATE_SENTINEL, 'w+').close()
 
+    @accepts(Dict('options', Bool('reboot', default=True)))
+    @job(lock='config_reset', logs=True)
+    def reset(self, job, options):
+        """
+        Reset database to configuration defaults.
+
+        If `reboot` is true this job will reboot the system after its completed with a delay of 10
+        seconds.
+        """
+        factorydb = f'{FREENAS_DATABASE}.factory'
+        if os.path.exists(factorydb):
+            os.unlink(factorydb)
+        cp = subprocess.run(
+            ['migrate93', '-f', factorydb],
+            capture_output=True,
+        )
+        if cp.returncode != 0:
+            job.logs_fd.write(cp.stderr)
+            raise CallError('Factory reset has failed.')
+        cp = subprocess.run(
+            [
+                'env', 'FREENAS_FACTORY=1', 'python', '/usr/local/www/freenasUI/manage.py',
+                'migrate', '--noinput', '--fake-initial',
+            ],
+            capture_output=True,
+        )
+        if cp.returncode != 0:
+            job.logs_fd.write(cp.stderr)
+            raise CallError('Factory reset has failed.')
+
+        subprocess.run(['mv', factorydb, FREENAS_DATABASE], check=True, capture_output=True)
```

Comment:
Why not `shutil.move`?

Reply:
No reason, it was just straight code move ; )